src/attr/_make.py

```diff
@@ -1873,64 +1874,136 @@ def _add_eq(cls, attrs=None):
 
 _already_repring = threading.local()
 
+if HAS_F_STRINGS:
+
+    def _make_repr(attrs, ns, cls):
+        unique_filename = "repr"
+        # Figure out which attributes to include, and which function to use to
+        # format them. The a.repr value can be either bool or a custom
+        # callable.
+        attr_names_with_reprs = tuple(
+            (a.name, repr if a.repr is True else a.repr, a.init)
+            for a in attrs
+            if a.repr is not False
+        )
+        globs = {
+            name + "_repr": r
+            for name, r, _ in attr_names_with_reprs
+            if r != repr
+        }
+        globs["_already_repring"] = _already_repring
+        globs["AttributeError"] = AttributeError
+        globs["NOTHING"] = NOTHING
+        attribute_fragments = []
+        for name, r, i in attr_names_with_reprs:
+            accessor = (
+                "self." + name
+                if i
+                else 'getattr(self, "' + name + '", NOTHING)'
+            )
+            fragment = (
+                "{name}={{{accessor}!r}}".format(name=name, accessor=accessor)
```

Comment:
look if you wanna optimize, optimize! 🤪

```suggestion
                "%s={%s!r}" % (name, accessor)
```